### DISCUSSION TITLE
Announcing Project Reunion V0.1!

### DISCUSSION THREAD

**Discussion by mevey** [microsoft/WindowsAppSDK#372] [Announcements]
Since the announcement of Project Reunion at Build this year, we’ve used this platform to directly engage and discuss features and issues that are important to you, the Windows developer community. We want to keep up the momentum by releasing an early build of Project Reunion to start the build pipeline and begin bringing in preview components. 

**So today, we are happy to announce the first Project Reunion preview!**  

V0.1 is just the beginning. This limited preview lays the foundation for Project Reunion’s runtime distribution and includes some of the work we’ve done towards creating the plumbing that makes sure your apps can consume Project Reunion APIs. _There’s still more to do_, to unify the platform, add more functionality such as Windowing and Storage APIs that will be released in the coming months, and do performance tuning. We will be publishing a more detailed roadmap in the new year, 2021. 

This preview is _not recommended for production use_, but try out the samples, experiment with the APIs, and give us feedback.  

### **Getting started with V0.1** 

Here are some documentation and samples available as part of this preview that you can try out, and most importantly, give feedback on. We encourage you to [ask questions, file bugs, and make proposals. ](https://github.com/microsoft/ProjectReunion/issues/new/choose)

- Project Reunion’s [**Hello World sample**](https://github.com/microsoft/Project-Reunion-Samples/tree/main/HelloWorld/reunioncppdesktopsampleapp) is a sandbox for you to test Project Reunion APIs without having to worry about the plumbing and setup, which includes a temporary workaround. It uses some [new App-Lifecycle APIs](https://github.com/microsoft/ProjectReunion/issues/111) that are in the works. 
- **DirectWriteCore** brings best-in-class text layout and rendering to your app and reduces the implementation cost across all Windows versions and other platforms because it is cross-platform and works downlevel. Checkout the [Font Viewer sample app](https://github.com/microsoft/Project-Reunion-Samples) to learn how you can utilize critical features, such as Variable Fonts, downlevel. 
- [**Modern Resources Toolkit (MRT) Core** ](https://github.com/microsoft/ProjectReunion/tree/main/dev/MRTCore)brings resource management innovation from the last twelve years to Win32 app developers. 

Other Project Reunion releases to check out: 

- WinUI 3 preview 3, which was released last month. This preview is currently packaged separately from Project Reunion V0.1. You can [learn how to use this preview to modernize your app UX ](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/)  
- [Microsoft Edge WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/) for Win32 C/C++, which is generally available and ready for use in production applications. 

Important links: 

- [Project Reunion NuGet package ](https://www.nuget.org/packages/Microsoft.ProjectReunion/)
- [Project Reunion Docs](https://docs.microsoft.com/windows/apps/project-reunion )
- [All Project Reunion samples ](https://github.com/microsoft/Project-Reunion-Samples)

Stay tuned for more announcements, previews, and releases!

**Comment by soumyamahunt** [comment]
The new project file format for Reunion sample app causing failure when using vcpkg packages, this is due to vcpkg detecting the project as uwp project instead of desktop app. I have already filed an issue in vcpkg repo [here](https://github.com/microsoft/vcpkg/issues/15409) also submitted [PR](https://github.com/microsoft/vcpkg/pull/15410) that should fix this issue for now. Since project reunion isn't finalized just yet and the format for project file may change in future, this issue needs some focus and should be discussed.

**Comment by Ronsss** [comment]
WHAT IS IT?
WHAT DOES IT DO?!
yes, i'm shouting.  

I wasn't at Build so I have no idea what this "reunion" thingie is.  I glean that it is some sort of programming language/environment ("hello world") but other than that ... "crickets" ...

**Reply by nickrandolph** [reply]
Have you read this: https://github.com/microsoft/ProjectReunion/blob/main/docs/README.md

**Reply by Ronsss** [reply]
Thanks, that answers my question. 

Looks interesting. But if it is only in V0.1, it still has a long way to go.

**Reply by nickrandolph** [reply]
Probably the more significant project to track right now is the WinUI project (https://docs.microsoft.com/en-us/windows/apps/winui/) and specifically the preview of WinUI3.

**Reply by Ronsss** [reply]
Thanks for the link to WinUI
.
So without a deep dive into both tools, it seems like MS  is repeating it's effort to unify development environments with 3 parallel tools: WinUI V2, V3 and Project Reunion.  What is the difference? I can see the possible progression from WinUI V2 to V3, but why should any developer waste time looking at vapourware like Project Reunion (at this time) ...?

**Reply by dotMorten** [reply]
PU is a little confusing to define. It's both its own project and a set of projects. Think of it this way that Project Reunion in itself is also part of what makes WinUI3 possible.
WinUi2 is just an extension to UWP. It's sort of misleading that they have the same name, as WinUI3 is WinUi2+ all the UI in UWP so it's so much more

**Reply by andrewleader** [reply]
Project Reunion is the "Windows Developer Platform" but shipped in a way that allows what we do to work down-level (so that you can use features in your app without requiring users to be on the very latest version of Windows), and in a way that supports more apps (like Win32 apps without requiring them to use MSIX).

Project Reunion actually includes WinUI 3. WinUI 3 is currently shipped separately, but going forward, WinUI 3 will be part of Project Reunion, along with other windowing APIs, file management APIs, and other platform APIs your apps use. WinUI will simply be one of the many features within Project Reunion.

Right now, v0.1 of Project Reunion is not something you'd use or find any benefit to (it's simply an engineering proof-of-concept that we could create a NuGet and ship a basic API in it). But with our upcoming releases over this year, you'll start to be able to do more, like use WinUI 3, etc.
